src/modules/nationalId/index.ts

```diff
@@ -2,30 +2,42 @@
  * Check National-id validation
  * @category National id
  * @method verifyIranianNationalId
- * @param  {String?}          nationalId [String of national id - like this: 1111111111]
+ * @param  {String?}          nationalId [String of national id - like this: 0018465986]
  * @return {Boolean}                    [valid or no]
  */
-function verifyIranianNationalId(nationalId?: string | number): boolean | null | undefined {
-	if (!nationalId) return;
-	else {
-		let code = nationalId.toString();
 
-		if (!code.match(/^\d{10}$/)) return false;
-		code = ("0000" + code).substr(code.length + 4 - 10);
+function verifyIranianNationalId(nationalId?: string | number): boolean | undefined {
+	if (!nationalId) return;
+	let code = nationalId.toString();
+	const codeLength = code.length;
+	const monoDigit = [
+		"0000000000",
+		"1111111111",
+		"2222222222",
+		"3333333333",
+		"4444444444",
+		"5555555555",
+		"6666666666",
+		"7777777777",
+		"8888888888",
+		"9999999999",
+	];
 
-		if (parseInt(code.substr(3, 6), 10) === 0) return false;
+	if (codeLength < 8 || codeLength > 10 || parseInt(code, 10) == 0) return false;
```

Comment:
There is no need to use `explicit coercion`

Reply:
The last condition checks the `monoDigit[0]` and this is duplicate and this is a DRY principle that should be removed.